src/lazy/components/server.tsx

```diff
@@ -8,6 +8,10 @@ export const createComponentServer = ({
   cacheId,
   dataLazyId,
 }: any) => (props: any) => {
+  if (ssr) {
```

Comment:
Isn't this just moving the logic slightly further down? I think this needs to be removed completely. Since we have the babel plugin to convert import to require, everything just works without doing anything additional.

Reply:
The babel plugin doesn't run during tests however. While I agree that we should not necessarily have this here, I would like to leave it in order to not have to significantly refactor how the integration tests work. 

One option would be to add `process.env.NODE_ENV === 'test'` or something to this conditional and then run the tests with this environment variable set.

Reply:
That doesn't seem to be the problem when running the tests. We already have a mock import setup in the failing integration tests that mimics the babel plugin behaviour. The root cause stems from using the deferred promise, requiring it to always be started to process anything. I think in SSR we should instead just pass in the loader, and execute it within the render if the `tryRequire` for the cache is not available.

i.e.

```
const LazyComponent: any = isServer
  ? createComponentServer({
      ssr,
      // Replace deferred, with: 
      loader,
      cacheId,
      dataLazyId,
    })
  : createComponentClient({
    ssr,
    defer,
    // Replace deferred, with:
    deferred: createDeferred(loader),
    cacheId,
    dataLazyId,
  });

...

const Resolved = ssr ? tryRequire(cacheId) || getExport(loader()) : null;
```

This should replicate existing server rendering behaviour during integration and fix the tests.

Reply:
Hmm okay nice catch I will try it